lib/vnet/ssh_handler.go

```diff
@@ -18,19 +18,39 @@ package vnet
 
 import (
 	"context"
+	"crypto/rand"
 	"net"
+	"strings"
+	"sync"
+	"time"
 
 	"github.com/gravitational/trace"
+	"github.com/jonboulle/clockwork"
+	"golang.org/x/crypto/ssh"
+
+	"github.com/gravitational/teleport/api/defaults"
+	"github.com/gravitational/teleport/api/utils/sshutils"
+)
+
+const (
+	// This host cert is generated on the fly and only trusted by SSH clients
+	// connecting locally to VNet addresses, the TTL doesn't matter very much.
+	hostCertTTL = defaults.MaxCertDuration
 )
 
 // sshHandler handles incoming VNet SSH connections.
 type sshHandler struct {
 	cfg sshHandlerConfig
+
+	hostCertMu      sync.Mutex
+	hostCert        ssh.Signer
+	hostCertExpires time.Time
```

Comment:
It takes like 30usec to sign an ed25519 cert, I think we're actually better off just signing a new one for each incoming connection.